docs/build/js-client/4-advanced-ipfs.md

```diff
@@ -0,0 +1,297 @@
+
+Initialize an empty `npm` package:
+
+```sh
+npm init
+```
+
+We will need these three packages for the application runtime: JS Client API, JS Client implementation for Node.js, and the package containing a well-maintained list of relay nodes.
+
+```sh
+npm install @fluencelabs/js-client.api @fluencelabs/js-client.node @fluencelabs/fluence-network-environment
+```
+
+Also, we need some specific packages for implementing IPFS service
+
+```sh
+npm install helia @helia/strings @helia/dag-json multiformats blockstore-fs datastore-fs
+```
+
+We will need the Fluence CLI to use the Aqua compiler, but only as a development dependency:
+
+```sh
+npm install --save-dev @fluencelabs/cli
+```
+
+Aqua comes with the standard library, which can be accessed from "@fluencelabs/aqua-lib" package. All Aqua packages are only needed at compiler time, so we install it as a development dependency:
+
+```sh
+npm install --save-dev @fluencelabs/aqua-lib
+```
+
+And last, but no least, we will need TypeScript:
+
+```sh
+npm install --save-dev typescript @types/node ts-node
+npx tsc --init
+```
+
+## Setting up Aqua compiler
+
+Let's put Aqua described earlier in the `aqua/files.aqua` file. You probably want to keep the generated TypeScript in the same directory with other typescript files, usually `src`. Let's create the `src/_aqua` directory for that.
+
+The overall project structure looks like this:
+
+```
+ ┣ aqua
+ ┃ ┗ files.aqua
+ ┣ src
+ ┃ ┣ _aqua
+ ┃ ┃ ┗ files.ts
+ ┃ ┗ index.ts
+ ┣ package-lock.json
+ ┣ package.json
+ ┗ tsconfig.json
+```
+
+To compile Aqua, we can use the Fluence CLI with `npm`:
+
+```sh
+npx fluence aqua -i ./aqua/ -o ./src/_aqua
+```
+
+To watch for Aqua file changes, add the `-w` flag:
+
+```sh
+npx fluence aqua -w -i ./aqua/ -o ./src/_aqua
+```
+
+We recommend to store this logic inside a script in the `packages.json` file:
+
+```javascript
+{
+  ...
+  "scripts": {
+    ...
+    "compile-aqua": "fluence aqua -i ./aqua/ -o ./src/_aqua",
+    "watch-aqua": "fluence aqua -w -i ./aqua/ -o ./src/_aqua"
+  },
+  ...
+}
+```
+
+## Using the compiled code in a TypeScript application
+
+Using the code generated by the compiler is as easy as calling a function. The compiler generates all the boilerplate needed to send a particle into the network and wraps it into a single call. It also generates a function for service callback registration. Note that all the type information and therefore type checking and code completion facilities are there!
+
+We will start with implementing `utils.ts` file:
+
+```typescript
+import { FsBlockstore } from 'blockstore-fs'; // Import block storage
+import { FsDatastore } from 'datastore-fs'; // Import app info storage
+import { createHelia } from 'helia'; // Lib for interacting with IPFS
+
+const blockstore = new FsBlockstore('./temp/store');
+const datastore = new FsDatastore('./temp/data');
+
+export async function makeHelia() { // Helper func for creating client node which will be used for reading IPFS
```

Comment:
`client node` is way too generic. Is it an IPFS node, or Fluence node or Node.js?